### PR TITLE
Form changing method for Shaymin Sky into Land at night

### PR DESCRIPTION
**Pull request by AsparagusEduardo**
## Description
Shaymin now reverts to Land Forme when at night.

## **Discord contact info**
AsparagusEduardo#6051

### CODE REVIEW

**Comment by AsparagusEduardo** [comment]
Put on hold until #1714 is merged

**Comment by AsparagusEduardo** [comment]
Merge conflicts resolved.
Also decided to refactor it in a similar way to `FORM_ITEM_USE_TIME` so it can support both day and night.

File: src/pokemon.c
Comment (BuffelSaft) [inline src/pokemon.c:8262]: I think it would be better to define constants for the day and night start/end times somewhere so that the same times are used for anything else that needs that information. IIRC the only other use is time based evolutions.

**Comment by BuffelSaft** [comment]
This is waaay overdue for a review. Needs conflicts resolved, but I'm happy to review it after that.

File: src/pokemon.c
Comment (AsparagusEduardo) [reply src/pokemon.c:8541]: I'll make a PR for pret, as I feel that would be better for it.

Comment (AsparagusEduardo) [reply src/pokemon.c:8541]: Actually, I would rather start defining those constants once we start looking into implementing DNS.

**Comment by AsparagusEduardo** [comment]
Conflicts solved. Ready for review.

**Comment by BuffelSaft** [comment]
I can't get this to work. I have a Shaymin sky in the party at 9:15am (which is apparently night time in Emerald) and it won't revert back to land forme. I've tried changing maps/map groups, battling and leaving the game running for a bit - no luck. Does it work on your end?

**Comment by DizzyEggg** [comment]
> I need to test it again, but I remember having it working but didn't immediately. Will test again.

Is it working now? 👀

**Comment by AsparagusEduardo** [comment]
> > I need to test it again, but I remember having it working but didn't immediately. Will test again.
> 
> Is it working now? 👀

I did not, in fact, test again 😂

I keep forgetting ^^;

**Comment by Bassoonian** [comment]
Reminder to test again (and also fix conflicts) :)

**Comment by AsparagusEduardo** [comment]
Issue solved, doing #3446 fixed the issue.

**Review by Bassoonian — APPROVED** [review]

**Comment by AsparagusEduardo** [comment]
To be more specific, the code I made was working, it's just that we weren't getting consistant results due to the issue described in #3446

**Comment by Bassoonian** [comment]
Finally confirmed that the code works and Shaymin Sky changes into Land at night. Waiting for CI to verify it and then I'll approve and merge this two year old PR :)

**Review by Bassoonian — APPROVED** [review]

**Pull request merged**
